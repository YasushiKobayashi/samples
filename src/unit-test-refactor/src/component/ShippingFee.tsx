import * as React from 'react'

import { isFreeShippingFee } from '@/models/cart'

interface Props {
  shippingFee: number
}

export const ShippingFee: React.FC<Props> = ({ shippingFee }) => {
  const label = isFreeShippingFee(shippingFee) ? '無料' : `${shippingFee.toLocaleString()}円`
  return <>{label}</>
}
