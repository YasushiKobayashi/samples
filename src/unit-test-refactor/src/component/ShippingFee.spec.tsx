import * as React from 'react'
import { cleanup, render } from '@testing-library/react'

import { ShippingFee } from './ShippingFee'

describe('ShippingFee', () => {
  afterEach(() => {
    cleanup()
  })

  it('送料0円の場合は無料と表示', () => {
    const { container, asFragment } = render(<ShippingFee shippingFee={0} />)
    expect(container.textContent).toContain('無料')
    expect(asFragment()).toMatchSnapshot()
  })

  it('送料500円の場合はそのまま表示', () => {
    const { container, asFragment } = render(<ShippingFee shippingFee={500} />)
    expect(container.textContent).toMatch(/500/)
    expect(asFragment()).toMatchSnapshot()
  })
})
