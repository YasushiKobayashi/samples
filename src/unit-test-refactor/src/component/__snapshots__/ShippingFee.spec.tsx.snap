// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ShippingFee > 送料0円の場合は無料と表示 1`] = `
<DocumentFragment>
  無料
</DocumentFragment>
`;

exports[`ShippingFee > 送料500円の場合はそのまま表示 1`] = `
<DocumentFragment>
  500円
</DocumentFragment>
`;

exports[`ShippingFee 送料0円の場合は無料と表示 1`] = `
<DocumentFragment>
  無料
</DocumentFragment>
`;

exports[`ShippingFee 送料500円の場合はそのまま表示 1`] = `
<DocumentFragment>
  500円
</DocumentFragment>
`;
